pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -902,3 +903,38 @@ func ValidateValidatingAdmissionPolicyUpdate(newC, oldC *admissionregistration.V
 func ValidateValidatingAdmissionPolicyBindingUpdate(newC, oldC *admissionregistration.ValidatingAdmissionPolicyBinding) field.ErrorList {
 	return validateValidatingAdmissionPolicyBinding(newC)
 }
+
+func validateValidatingAdmissionPolicyStatus(status *admissionregistration.ValidatingAdmissionPolicyStatus, fldPath *field.Path) field.ErrorList {
+	return validateTypeChecking(status.TypeChecking, fldPath.Child("typeChecking"))
+}
+
+func validateTypeChecking(typeChecking *admissionregistration.TypeChecking, fldPath *field.Path) field.ErrorList {
+	if typeChecking == nil {
+		return nil
+	}
+	return validateExpressionWarnings(typeChecking.ExpressionWarnings, fldPath.Child("expressionWarnings"))
+}
+
+func validateExpressionWarnings(expressionWarnings []admissionregistration.ExpressionWarning, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	for i, warning := range expressionWarnings {
+		allErrors = append(allErrors, validateExpressionWarning(&warning, fldPath.Index(i))...)
+	}
+	return allErrors
+}
+
+func validateExpressionWarning(expressionWarning *admissionregistration.ExpressionWarning, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	if expressionWarning.Expression == "" {
+		allErrors = append(allErrors, field.Required(fldPath.Child("expression"), "expression must be non-empty if specified"))
+	}
+	if expressionWarning.Warning == "" {
+		allErrors = append(allErrors, field.Required(fldPath.Child("warning"), "warning must be non-empty if specified"))
+	}
+	if expressionWarning.RefField == "" {
+		allErrors = append(allErrors, field.Required(fldPath.Child("refField"), "refField must be non-empty if specified"))
+	} else if expressionWarning.RefField != admissionregistration.ExpressionRefFieldValidation {
+		allErrors = append(allErrors, field.NotSupported(fldPath.Child("refField"), expressionWarning.RefField, []string{admissionregistration.ExpressionRefFieldValidation}))
+	}
```

Comment:
same comment about it not being clear what the allowed values for this are and how they are to be used

if this is an enum we expect to expand in the future, need to tolerate existing persisted unknown values when validating updates (so a new server that expands allowed values can persist data we can tolerate in a no-op update against an older server)

Reply:
(obsolete)